demos/rain_prediction.ipynb

```diff
@@ -0,0 +1,467 @@
+    "\n",
+    "However, we will skip a lot of possible feature engineering etc. in order to focus on the application of LakeFS and LakeFS-spec. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import json \n",
+    "\n",
+    "def transform_json_weather_data(filepath):\n",
+    "    f = open(filepath)\n",
+    "    data = json.load(f)\n",
+    "    f.close()\n",
+    "\n",
+    "    df = pd.DataFrame.from_dict(data[\"hourly\"])\n",
+    "    df.time = pd.to_datetime(df.time)\n",
+    "    df['is_raining'] = df.rain > 0\n",
+    "    df['is_raining_in_1_day'] = df.is_raining.shift(24)\n",
+    "    df = df.dropna()\n",
+    "    return df\n",
+    "    \n",
+    "df = transform_json_weather_data('./data/weather-2010s.json')\n",
+    "df.head(5)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let us now save this data as a csv into the main branch. The `.to_csv` method calls a `put` operation behind the scenes, our commit hook is called and the file committed. You can verify the saving worked in your LakeFS GUI in the browser when looking at the Comits and uncommitted changes tabs of the main branch. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df.to_csv(f'lakefs://{REPO_NAME}/main/weather_2010s.csv')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will now do a train test split."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sklearn.model_selection\n",
+    "\n",
+    "model_data=df.drop('time', axis=1)\n",
+    "\n",
+    "train, test = sklearn.model_selection.train_test_split(model_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We save these train and test datasets into a new `training` branch. If, as in this case, the branch does not yet exist, it is implicitly created. You can steer this behaviour via the `create_branch_ok` flag when initializing the `LakeFSFileSystem`. The flag defaults to `True`.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TRAINING_BRANCH = 'training'\n",
+    "train.to_csv(f'lakefs://{REPO_NAME}/{TRAINING_BRANCH}/train_weather.csv')\n",
+    "test.to_csv(f'lakefs://{REPO_NAME}/{TRAINING_BRANCH}/test_weather.csv')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The implicit branch creation is a convenient way to create new branches programmatically. However, one drawback is that typos might result in new branch creation. If you want these to throw errors instead, disable the implicit branch creation"
```

Comment:
```suggestion
    "Implicit branch creation is a convenient way to create new branches programmatically. However, one drawback is that typos in your code might result in new accidental branch creations. If you want to avoid this implicit behavior and raise errors instead, you can disable implicit branch creation by setting `fs.create_branch_ok=False`."
```